src/main/controller/PromotionController.ts

```diff
@@ -103,23 +111,37 @@ export class PromotionController {
           { abortEarly: false }
         );
 
-        // find existing restaurant
-        const restaurant = await transactionalEntityManager
-          .getCustomRepository(RestaurantRepository)
-          .findOne({ placeId: promotionDTO.placeId });
         const user = await transactionalEntityManager
           .getCustomRepository(UserRepository)
           .findOneOrFail(promotionDTO.userId);
+
+        // find existing restaurant
+        let restaurant = await transactionalEntityManager
+          .getCustomRepository(RestaurantRepository)
+          .findOne({ placeId: promotionDTO.placeId });
+
+        if (!restaurant) {
+          const coordinates = await this.geocodingService.getGeoCoordinatesFromAddress(
+            promotionDTO.googlePlacesAddress
+          );
+          if (coordinates.lat && coordinates.lon) {
+            restaurant = new Restaurant(
+              promotionDTO.placeId,
+              coordinates.lat,
+              coordinates.lon
+            );
+          } else {
+            restaurant = new Restaurant(promotionDTO.placeId);
```

Comment:
Here, we'll save the restaurant regardless if we're unable to fetch lat/lon values! FE will deal with whether or not lat/lon are null